openedxscorm/static/js/src/scormxblock.js

```diff
@@ -177,27 +177,27 @@ function ScormXBlock(runtime, element, settings) {
         "cmi.completion_status",
         "cmi.success_status",
         "cmi.core.score.raw",
-        "cmi.score.raw"
+        "cmi.score.raw",
+        "cmi.mode"
+        
     ];
     var getValueUrl = runtime.handlerUrl(element, 'scorm_get_value');
     var GetValue = function (cmi_element) {
         // Only make a call if navigation menu was not used
         // Otherwise the synchronous calls are blocked by chromium on page unload
-        if (uncachedValues.includes(cmi_element) && !navigationClick) {
-            $.ajax({
+        if (uncachedValues.includes(cmi_element) && !navigationClick){
+            var response = $.ajax({
                 type: "POST",
                 url: getValueUrl,
                 data: JSON.stringify({
-                    'name': cmi_element
+                    'name': cmi_element,
+                    'url': window.location.href
                 }),
-                async: false,
-                success: function (response) {
```

Comment:
why we have to remove this `success` handler?

Reply:
Unable to retrieve user mode through **self.runtime**. Although runtime has a **handleurl** function , isn't returning correct URLs.

Reply:
Synchronous AJAX call within GetValue does not properly return the response value in success handler, resulting in an empty string. This causes GetValue to be called twice, so the mode is not being set in this case.

Reply:
I'm not getting it why synchronous AJAX call within GetValue does not properly return the response value?